pages/highscores/index.js

```diff
@@ -0,0 +1,51 @@
+import useHighscoresStore from "../../src/zustand/useHighscoresStore.js";
+import { nanoid } from "nanoid";
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+
+const StyledContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  max-height: 100vh;
+  width: 100vw;
+  overflow-y: scroll;
+`;
+
+const StyledHighscoresList = styled.ul`
+  list-style: none;
+  max-width: 440px;
+  width: 80vw;
+  padding: 0;
+`;
+
+const StyledListItem = styled.li`
+  border: 2px solid black;
+  padding: 7px 7px 7px 15px;
+  margin: 10px 0;
+`;
+
+export default function Highscores() {
+  const { highscores } = useHighscoresStore();
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(false);
+  }, []);
+
+  if (isLoading) {
+    return <p>isLoading</p>;
+  }
+  return (
+    <StyledContainer>
+      <h1>Highscores</h1>
+      <StyledHighscoresList>
+        {highscores.map((highscore) => {
+          return (
+            <StyledListItem key={nanoid()}>{highscore.name}</StyledListItem>
```

Comment:
On every render, a completely different key is generated by `nanoid()`; this is not what React needs. Please pass a unique, but steady key.

Reply:
I am not 100% sure if I understand this right: 
I tried to fix it now by setting the `nanoid` directly in the zustand-store (`useHighscoresStore`):

```
const useHighscoresStore = create(
  persist(
    (set) => ({
      highscores: [],
      addHighscore: (name, time, failed) =>
        set((state) => ({
          highscores: [
            ...state.highscores,
            { name: name, time: time, failed: failed, id: nanoid() },
          ],
        })),
    }),
    {
      name: "highscores",
    }
  )
);
```

Then I can set the key to `key=highscore.id`:
```
      <StyledHighscoresList role="list">
        {highscores.map((highscore) => {
          return (
            <StyledListItem key={highscore.id}>{highscore.name}</StyledListItem>
          );
        })}
      </StyledHighscoresList>
```

Is this the right approach?